### ISSUE TITLE
Setting cookies inside GraphQL resolvers

### ISSUE THREAD

**Issue by FOSSforlife** [goldcaddy77/warthog#361]
I've been attempting to use [this](https://github.com/benawad/jwt-auth-example/tree/master/server) tutorial (written using TypeORM and TypeGraphQL) as a guide for setting up authentication. He uses a "refresh token" method which he explains in [this](https://www.youtube.com/watch?v=iD49_NIQ-R4) video, which is supposed to be more secure than relying just on JWT or cookies because it makes use of both.

In his code, he has a [login](https://github.com/benawad/jwt-auth-example/blob/master/server/src/UserResolver.ts#L83) mutation which, upon success, sets a refresh token using Express's `res.cookie()`. The problem with implementing this in Warthog is that the login mutation does not have access to the `res` object, as Warthog's context object only passes in  `req`. The only place we can access `res`, as @goldcaddy77 has suggested, is inside the [onBeforeGraphQLMiddleware and onAfterGraphQLMiddleware](https://github.com/goldcaddy77/warthog/blob/a02a237b58269fd22b219c31b7a106a4d42af815/src/core/server.ts#L47) hooks. I'm just not sure how to get these hooks to run specifically when a user is logging in, since that's when I need to set the refresh token. 

I also had the idea of passing `res` into the context object using the onBeforeGraphQLMiddleware hook, so that I could then access it from within GraphQL like the example shows, but this function doesn't seem to have access to the context object. 

It doesn't seem like I'll be able to replicate the example I was looking at using Warthog, and if that's the case, I'd like to know the best solution for setting cookies inside GraphQL resolvers.

**Comment by goldcaddy77** [comment]
Gotcha! So it could be as simple as passing `res` in context as well?

https://github.com/goldcaddy77/warthog/blob/master/src/core/server.ts#L228

**Comment by goldcaddy77** [comment]
If that’s the case, feel free to open up a PR or I can get to it this week

**Comment by FOSSforlife** [comment]
Yep! I'll go ahead and attempt implementing it myself :) Hopefully it's as straight forward as it seems.

**Comment by goldcaddy77** [comment]
Awesome. If you tackle this, let’s keep `request` there for backwards compatibility and additionally add `req` and `res`.

**Comment by FOSSforlife** [comment]
Right, I forgot that `request` was named as such here instead of `req`. In that case, I'll just do `request` and `response` if that works with you.

**Comment by goldcaddy77** [comment]
I’d prefer `req` and `res` to indicate they’re the express req and res objects.

**Comment by FOSSforlife** [comment]
Sorry for not making progress on this yet, I keep getting super wrapped up in work and other priorities. I should have time this weekend, but if anyone else wants to take care of it sooner I'm also fine with that. Otherwise I'll get to it on saturday.

**Comment by goldcaddy77** [comment]
No worries, I’ll review once I see a PR come through

**Comment by FOSSforlife** [comment]
Hey @goldcaddy77, sorry for not keeping you posted. I had started to make progress on this, but didn't have anything working. I don't think I'll be able to accomplish this task, as I've got many other things going on and have currently de-prioritized my project which was using Warthog. Hope you understand and sorry to disappoint.

**Comment by goldcaddy77** [comment]
No worries - I'll take a crack at this when I get through the other items on my list.
